crates/corro-agent/src/agent/util.rs

```diff
@@ -0,0 +1,1479 @@
+                    Ok(true) => {
+                        debug!(%actor_id, %version, "succesfully applied buffered changes");
+                    }
+                    Err(e) => {
+                        error!(%actor_id, %version, "could not apply fully buffered changes: {e}");
+                    }
+                }
+            }
+        }
+    }
+}
+
+// DOCME: provide some context for this function
+// TODO: move to a more appropriate module?
+pub async fn clear_buffered_meta_loop(
+    agent: Agent,
+    mut rx_partials: Receiver<(ActorId, RangeInclusive<Version>)>,
+) {
+    while let Some((actor_id, versions)) = rx_partials.recv().await {
+        let pool = agent.pool().clone();
+        let self_actor_id = agent.actor_id();
+        tokio::spawn(async move {
+            loop {
+                let res = {
+                    let mut conn = pool.write_low().await?;
+
+                    block_in_place(|| {
+                        let tx = conn.immediate_transaction()?;
+
+                        // TODO: delete buffered changes from deleted sequences only (maybe, it's kind of hard and may not be necessary)
+
+                        let seq_count = tx
+                            .prepare_cached("DELETE FROM __corro_seq_bookkeeping WHERE (site_id, version, start_seq) IN (SELECT site_id, version, start_seq FROM __corro_seq_bookkeeping WHERE site_id = ? AND version >= ? AND version <= ? LIMIT ?)")?
+                            .execute(params![actor_id, versions.start(), versions.end(), TO_CLEAR_COUNT])?;
+
+                        let buf_count = tx
+                            .prepare_cached("DELETE FROM __corro_buffered_changes WHERE (site_id, db_version, version, seq) IN (SELECT site_id, db_version, version, seq FROM __corro_buffered_changes WHERE site_id = ? AND version >= ? AND version <= ? LIMIT ?)")?
+                            .execute(params![actor_id, versions.start(), versions.end(), TO_CLEAR_COUNT])?;
+
+                        tx.commit()?;
+
+                        Ok::<_, rusqlite::Error>((buf_count, seq_count))
+                    })
+                };
+
+                match res {
+                    Ok((buf_count, seq_count)) => {
+                        if buf_count + seq_count > 0 {
+                            info!(%actor_id, %self_actor_id, "cleared {} buffered meta rows for versions {versions:?}", buf_count + seq_count);
+                        }
+                        if buf_count < TO_CLEAR_COUNT && seq_count < TO_CLEAR_COUNT {
+                            break;
+                        }
+                    }
+                    Err(e) => {
+                        error!(%actor_id, "could not clear buffered meta for versions {versions:?}: {e}");
+                    }
+                }
+
+                tokio::time::sleep(Duration::from_secs(2)).await;
+            }
+
+            Ok::<_, eyre::Report>(())
+        });
+    }
+}
+
+// DOCME: provide some context for this function
```

Comment:
When we receive an empty changeset or when we find them by periodically calling the `find_cleared_db_versions` function, we immediately apply them to our in-memory state, but we don't do it in the DB. This is to reduce DB load and be more efficient by chunking these operations.